### PR TITLE
Nullable reference type annotations part 5

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

File: src/GraphQL/Utilities/Federation/AnyValue.cs [hunk 1]
```diff
@@ -1,10 +1,12 @@
+#nullable enable
+
 using GraphQL.Language.AST;
 
 namespace GraphQL.Utilities.Federation
 {
     public class AnyValue : ValueNode<object>
     {
-        public AnyValue(object value) : base(value)
+        public AnyValue(object? value) : base(value!)
```
Comment (sungam3r) [inline src/GraphQL/Utilities/Federation/AnyValue.cs:9]: )

File: src/GraphQL/Utilities/SchemaBuilder.cs [hunk 2]
```diff
@@ -178,7 +180,7 @@ private Schema BuildSchemaFrom(GraphQLDocument document)
             }
             else
             {
-                schema.Query = GetType("Query") as IObjectGraphType;
+                schema.Query = (GetType("Query") as IObjectGraphType)!;
```
Comment (sungam3r) [inline src/GraphQL/Utilities/SchemaBuilder.cs:183]: schema.Query = (GetType("Query") as IObjectGraphType)!;
->
 schema.Query = (IObjectGraphType)GetType("Query");

it is easier

File: src/GraphQL/Utilities/AstPrinter.cs [hunk 3]
```diff
@@ -31,12 +33,17 @@ public static string Print(INode node)
...
+    internal abstract class AstPrintConfig
     {
         internal List<AstPrintFieldDefinition> FieldsList { get; } = new List<AstPrintFieldDefinition>();
         public IEnumerable<AstPrintFieldDefinition> Fields => FieldsList;
-        public Func<INode, bool> Matches { get; set; }
-        public Func<IDictionary<string, object>, object> PrintAst { get; set; }
+        public Func<INode, bool> Matches { get; }
+        public Func<IDictionary<string, object?>, object>? PrintAst { get; set; }
+
+        public AstPrintConfig(Func<INode, bool> matches)
```
Comment (sungam3r) [inline src/GraphQL/Utilities/AstPrinter.cs:43]: protected maybe

File: src/GraphQL/Utilities/SchemaPrinter.cs [hunk 4]
```diff
@@ -324,11 +326,11 @@ public string FormatDefaultValue(object value, IGraphType graphType)
...
             return graphType switch
             {
-                NonNullGraphType nonNull => FormatDefaultValue(value, nonNull.ResolvedType),
-                ListGraphType list => "[{0}]".ToFormat(string.Join(", ", ((IEnumerable<object>)value).Select(i => FormatDefaultValue(i, list.ResolvedType)))),
+                NonNullGraphType nonNull => FormatDefaultValue(value, nonNull.ResolvedType!),
+                ListGraphType list => "[{0}]".ToFormat(string.Join(", ", ((IEnumerable<object>)value).Select(i => FormatDefaultValue(i, list.ResolvedType!)))),
                 IInputObjectGraphType input => FormatInputObjectValue(value, input),
-                EnumerationGraphType enumeration => (enumeration.Serialize(value) ?? throw new ArgumentOutOfRangeException(nameof(value), $"Unknown value '{value}' for enumeration '{enumeration.Name}'")).ToString(),
-                ScalarGraphType scalar => AstPrinter.Print(scalar.ToAST(value)),
+                EnumerationGraphType enumeration => AstPrinter.Print(enumeration.ToAST(value) ?? throw new ArgumentOutOfRangeException(nameof(value), $"Unable to convert '{value}' to AST for enumeration type '{enumeration.Name}'.")),
```
Comment (sungam3r) [inline src/GraphQL/Utilities/SchemaPrinter.cs:332]: bugfix?

Comment (Shane32) [reply src/GraphQL/Utilities/SchemaPrinter.cs:332]: Yes. Pretty obvious, considering the new scalar design.

File: src/GraphQL/Utilities/Federation/AnyValue.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL/Utilities/Federation/AnyValue.cs:9]: Maybe we can change the base type to `ValueNode<object?>` which could be more appropriate. I really don’t know how AnyValue is used so I was probably trying to make as few changes as possible.

**Pull request merged**